### PR TITLE
Fix download exception handling

### PR DESCRIPTION
**Pull request by fviernau**
The Downloader may crash due to setting the cause of a throwable twice.
Adjust the exception handling to avoid this, see individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1365)

### CODE REVIEW

File: downloader/src/funTest/kotlin/DownloaderTest.kt [hunk 1]
```diff
@@ -106,8 +106,8 @@ class DownloaderTest : StringSpec() {
                 Downloader().download(pkg, outputDir)
             }
 
-            exception.cause shouldNotBe null
-            exception.cause!!.message shouldBe "Source artifact does not match expected SHA-1 hash " +
+            exception.suppressed.size shouldBe 3
+            exception.suppressed.get(1)!!.message shouldBe "Source artifact does not match expected SHA-1 hash " +
```
Comment (mnonnenmacher) [inline downloader/src/funTest/kotlin/DownloaderTest.kt]: If we expect three suppressed exceptions we should verify that all of them are correct, not only the second.

Comment (fviernau) [reply downloader/src/funTest/kotlin/DownloaderTest.kt]: I intentionally did not do that in the first as I considered it out of scope of this commit and of this test as well. Anyhow, it seems good to have it covered, addressed in a separate commit.

File: downloader/src/main/kotlin/Downloader.kt [hunk 2]
```diff
@@ -139,7 +139,7 @@ class Downloader {
             "The output directory '$targetDir' must not contain any files yet."
         }
 
-        var previousException: DownloadException? = null
+        val exception = DownloadException("Download failed for '${target.id.toCoordinates()}'.")
```
[the remark this answers is not in the document]
Comment (fviernau) [reply downloader/src/main/kotlin/Downloader.kt]: done

**Comment by sschuberth** [comment]
@fviernau, please rebase this onto *master* once https://github.com/heremaps/oss-review-toolkit/pull/1373 is merged. That should make it easier to adjust the `ExcelReporterTest` to make it pass for this PR.

File: reporter/src/funTest/kotlin/reporters/ExcelReporterTest.kt [hunk 3]
```diff
@@ -40,7 +40,7 @@ class ExcelReporterTest : WordSpec({
             .readValue<OrtResult>()
 
     "ExcelReporter" should {
-        "successfully export to an Excel sheet" {
+        "successfully export to an Excel sheet".config(enabled = false) {
```
Comment (mnonnenmacher) [inline reporter/src/funTest/kotlin/reporters/ExcelReporterTest.kt]: Commit message: "tetst" -> "test".

Comment (sschuberth) [reply reporter/src/funTest/kotlin/reporters/ExcelReporterTest.kt]: Done.

**Pull request merged**